### ISSUE TITLE
Access token header without obtaining payload?

### ISSUE THREAD

**Issue by naerbnic** [Keats/jsonwebtoken#36]
In order to provide an accurate key for some applications (e.g. OpenID Connect for Google), we need to be able to access the kid (Key ID) field of the header. I could also imagine other fields in the header would be necessary to find the validation context.

The current API allows by decoding a JWT with no signature verification, however I feel that it would be less error-prone if there were a separate function to access the header directly. In that case, you wouldn't have a non-validated payload in the code, and have less chance of accidentally using it without appropriate validation.

The simplest way would be just to add a function `decode_header` like so:

```rust
fn decode_header(token: &str) -> Result<header::Header> ...
```

**Comment by Keats** [comment]
That's a good point, it would require a major version change though to remove the option to do no-sig decoding.
I'll take a PR for that or do it when I have time

**Comment by Keats** [comment]
Fixed in 3.0
